Purchases/Caching/RCInMemoryCachedObject.m

```diff
@@ -0,0 +1,70 @@
+//
+//  RCInMemoryCachedObject.m
+//  Purchases
+//
+//  Created by RevenueCat.
+//  Copyright © 2020 Purchases. All rights reserved.
+//
+#import <Foundation/Foundation.h>
+
+#import "RCInMemoryCachedObject.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+
+@interface RCInMemoryCachedObject ()
+
+@property (nonatomic, nullable) NSDate *lastUpdatedAt;
+@property (nonatomic, assign) int cacheDurationInSeconds;
+@property (nonatomic, nullable) id cachedInstance;
+
+@end
+
+
+NS_ASSUME_NONNULL_END
+
+
+@implementation RCInMemoryCachedObject
+
+- (instancetype)initWithCacheDurationInSeconds:(int)cacheDurationInSeconds {
+    return [self initWithCacheDurationInSeconds:cacheDurationInSeconds
+                                  lastUpdatedAt:nil];
+}
+
+- (instancetype)initWithCacheDurationInSeconds:(int)cacheDurationInSeconds
+                                 lastUpdatedAt:(NSDate *_Nullable)lastUpdatedAt {
+    if (self == [super init]) {
+        self.cacheDurationInSeconds = cacheDurationInSeconds;
+        self.lastUpdatedAt = lastUpdatedAt;
+    }
+    return self;
+}
+
+- (BOOL)isCacheStale {
+    if (self.lastUpdatedAt == nil) {
+        return YES;
+    }
+
+    NSTimeInterval timeSinceLastCheck = -1 * [self.lastUpdatedAt timeIntervalSinceNow];
+    return timeSinceLastCheck >= self.cacheDurationInSeconds;
+}
+
+- (void)clearCacheTimestamp {
+    self.lastUpdatedAt = nil;
+}
+
+- (void)clearCache {
+    [self clearCacheTimestamp];
+    self.cachedInstance = nil;
+}
+
+- (void)cacheInstance:(id)instance date:(NSDate *)date {
```

Comment:
should this parameters be nonnull? That's why we have `clearCache` right?

Reply:
good catch. I meant to do `NS_ASSUME_NONNULL` for the whole thing, not just the interface.

Reply:
5eece5206b24678086f62a9acafa1bc04a6f7543